atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/service/TransactionStatus.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.service;
+
+import org.derive4j.Data;
+
+@Data
+public interface TransactionStatus {
+    interface Cases<R> {
+        R inProgress();
+
+        R committed(long commitTimestamp);
+
+        R aborted();
+
+        R unknown();
+    }
```

Comment:
Maybe overkill, but I would recommend having some docs here around what inProgress and especially unknown mean